src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -292,6 +302,47 @@ private bool IsManifestIdOutdated(string workloadManifestDir)
             return _outdatedManifestIds.Contains(manifestId);
         }
 
+        private Dictionary<string, WorkloadSet> GetAvailableWorkloadSets()
+        {
+            Dictionary<string, WorkloadSet> availableWorkloadSets = new Dictionary<string, WorkloadSet>();
+
+            foreach (var manifestRoot in _manifestRoots.Reverse())
+            {
+                //  Workload sets must match the SDK feature band, we don't support any fallback to a previous band
+                var workloadSetsRoot = Path.Combine(manifestRoot, _sdkVersionBand.ToString(), WorkloadSetsFolderName);
+                if (Directory.Exists(workloadSetsRoot))
+                {
+                    foreach (var workloadSetDirectory in Directory.GetDirectories(workloadSetsRoot))
+                    {
+                        WorkloadSet? workloadSet = null;
+                        foreach (var jsonFile in Directory.GetFiles(workloadSetDirectory, "*.workloadset.json"))
+                        {
+                            var newWorkloadSet = WorkloadSet.FromJson(File.ReadAllText(jsonFile), _sdkVersionBand);
+                            if (workloadSet == null)
+                            {
+                                workloadSet = newWorkloadSet;
+                            }
+                            else
+                            {
+                                //  If there are multiple workloadset.json files, merge them
+                                foreach (var kvp in newWorkloadSet.ManifestVersions)
+                                {
+                                    workloadSet.ManifestVersions.Add(kvp.Key, kvp.Value);
```

Comment:
I think it might be possible for a duplicate to be added here if it was on the disk, would that be a problem? Should we do any validation here?

Reply:
It will throw an exception:

> System.ArgumentException : An item with the same key has already been added. Key: ios

I've added a test for this.

The exception is not very helpful but this is a scenario that may never be used.  I expect all the workload sets we create to just have one json file.  The reason for supporting multiple is to leave the opportunity for something like Tizen to augment a workload set with versions for additional manifests.